client/core/core.go

```diff
@@ -2219,6 +2219,12 @@ func handleRevokeMatchMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) erro
 		return fmt.Errorf("no order found with id %s", oid.String())
 	}
 
+	// todo: consider checking if any refunds/forced redemption are necessary
+	// before setting this match's status to Revoked.
+	// Alternatively, set status to Revoked but "remember" to attempt Refund/
+	// FindRedemption if necessary. Currently, `trackedTrade.isRefundable` will
+	// return false for this match because of the status change.
+
 	md := tracker.metaData
 	md.Status = order.OrderStatusRevoked
```

Comment:
Hmm. I now see that there are some issues here with revocation handling. Revocations can apply to individual matches and not entire order. The server side handling has some of the logic conflated as well, it seems, where the revocation is applied to the order instead of the match. If I'm reading this right, it's going to take some work to get it corrected. For now, I think we should handle it by tracking the revocation as part of the `MatchProof`. 

Though it's tempting, we don't want to add a new `order.MatchStatus` e.g. `MatchRevoked`, because we may still want to know what the order status was when it was revoked.

Reply:
I think we resolved the server side concerns in matrix (order revocation was a side effect of failing a match, but the individual match was being revoked via `SetMatchInactive`), but please confirm if we need to address something on that end.

Reply:
Oh, I agree we should add a new `order.MatchStatus` value for failed/revoked.  On the server DB implementation, there are `active BOOLEAN` and `status INT2` columns where a revoked/failed match is identifiable by `NOT active AND status != {MatchComplete flag}`.

And the reason for revoking an order belonging to the party at fault in a failing swap is that if the user is having trouble completing swaps we do not want the order to stay on the books.

Reply:
grr, I mean't should **not** add a new status value.

Reply:
> I think we resolved the server side concerns in matrix (order revocation was a side effect of failing a match, but the individual match was being revoked via `SetMatchInactive` and removed from the `Swapper.matches` map after requesting `'revoke'` acks from both users), but please confirm if we need to address something on that end.

Yeah, server side checks out as discussed in chat. Don't think there's any issue there.